### PR TITLE
Quote() constness and cleanup

### PR DESCRIPTION
**Pull request by bmorel**
Quote() function iterates over 4 buffers for no reason. Get rid of that.
Also use a std::string to store the cached data, which prevents to meet too long strings.
Last, make it's returned value const, as it should.

### CODE REVIEW

**Comment by slipher** [comment]
Multiple buffers are needed in places such as this.
```
			ADMP( va( "%s %s %s %d", QQ( N_("^3builder:^* $1$ built by $2$^*, buildlog #$3$") ),
				  Quote( buildingName ), Quote( builder ), MAX_CLIENTS + level.buildId - i - 1 ) );
```

**Comment by slipher** [comment]
> As in, the order in which parameters expression are resolved is up to the compiler to decide.

That's correct. But the code above still works correctly regardless of which one is evaluated first.

**Comment by bmorel** [comment]
This is quite dangerous though, as in, it means you can't call Quote() more than 4 times this way, and there's nothing explicitly saying that limitation or other safeguard.

I wonder if we could improve on this?

File: src/sgame/sg_utils.cpp [hunk 1]
```diff
@@ -943,15 +943,17 @@ std::string G_EscapeServerCommandArg( Str::StringRef str )
 // Escape a command for used in server commands (sent from client to server)
 // Difference from Cmd::Escape and normal command parsing is that newlines are allowed
 // (for commands that have multi-line output)
-char *Quote( Str::StringRef str )
+// WARNING: this function can only be called BUFFERSIZE times by the same function,
+// or bad things will happen!
+const char *Quote( Str::StringRef str )
 {
-	static char buffer[ 4 ][ MAX_STRING_CHARS ];
+	int constexpr BUFFERSIZE = 256; // must be a power of 2
```
Comment (slipher) [inline src/sgame/sg_utils.cpp:950]: Surely 8 or 16 is enough? Also BUFFERSIZE is not a good name as this is a number of buffers, not the size of a buffer.

Comment (bmorel) [reply src/sgame/sg_utils.cpp:950]: I wrote this after talking with @cu-kai on IRC, and it seems he needs to run this function in a loop with MAX_CLIENTS iterations.

> Also BUFFERSIZE

Have to admit I didn't took much time at finding a variable name. Initially I wanted to get rid of that limit entirely, perhaps using a vector, but my short tinkering didn't let me find a satisfying solution (looked uglier than current).

Tbh I wonder how much this stuff makes sense.

Comment (slipher) [reply src/sgame/sg_utils.cpp:950]: You should never store the pointer returned by this in a variable. The max limit should just accommodate when you use several of them in a single line like the print(%s %s %s) things where each argument is quoted.

**Comment by bmorel** [comment]
superseeded by #2735

**Pull request closed without merging**